docs/en/integrations/data-ingestion/apache-spark/spark-native-connector.md

```diff
@@ -0,0 +1,583 @@
+---
+sidebar_label: Spark Native Connector
+sidebar_position: 2
+slug: /en/integrations/apache-spark/spark-native-connector
+description: Introduction to Apache Spark with ClickHouse
+keywords: [ clickhouse, apache, spark, migrating, data ]
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import TOCInline from '@theme/TOCInline';
+
+# Spark Connector
+
+This connector leverages ClickHouse-specific optimizations, such as advanced partitioning and predicate pushdown, to
+improve query performance and data handling.
+The connector is based on [ClickHouse's official JDBC connector](https://github.com/ClickHouse/clickhouse-java), and
+manages its own catalog.
+
+Before Spark 3.0, Spark lacked a built-in catalog concept, so users typically relied on external catalog systems such as Hive Metastore or AWS Glue.
+With these external solutions, users had to register their data source tables manually before accessing them in Spark.
+However, since Spark 3.0 introduced the catalog concept, Spark can now automatically discover tables by registering catalog plugins.
+
+Spark default catalog is `spark_catalog`, and tables are identified by `{catalog name}.{database}.{table}`. With the new
+catalog feature, it is now possible to add and work with multiple catalogs in a single Spark application.
+
+
+<TOCInline toc={toc}></TOCInline>
+
+## Requirements
+
+- Java 8 or 17
+- Scala 2.12 or 2.13
+- Apache Spark 3.3 or 3.4 or 3.5
+
+## Compatibility Matrix
+
+| Version | Compatible Spark Versions | ClickHouse JDBC version |
+|---------|---------------------------|-------------------------|
+| main    | Spark 3.3, 3.4, 3.5       | 0.6.3                   |
+| 0.8.0   | Spark 3.3, 3.4, 3.5       | 0.6.3                   |
+| 0.7.3   | Spark 3.3, 3.4            | 0.4.6                   |
+| 0.6.0   | Spark 3.3                 | 0.3.2-patch11           |
+| 0.5.0   | Spark 3.2, 3.3            | 0.3.2-patch11           |
+| 0.4.0   | Spark 3.2, 3.3            | Not depend on           |
+| 0.3.0   | Spark 3.2, 3.3            | Not depend on           |
+| 0.2.1   | Spark 3.2                 | Not depend on           |
+| 0.1.2   | Spark 3.2                 | Not depend on           |
+
+
+## Installation & Setup
+
+For integrating ClickHouse with Spark, there are multiple installation options to suit different project setups.
+You can add the ClickHouse Spark connector as a dependency directly in your project’s build file (such as in `pom.xml` for Maven or `build.sbt` for SBT).
```

Comment:
Maybe we should point to our examples [directory](https://github.com/ClickHouse/spark-clickhouse-connector/tree/main/examples/scala)

Reply:
We talked about it - as a rule of thumb, we want to keep the code examples within our docs and not in each repo. I copied the `sbt` example to be included here.